borg/helpers.py

```diff
@@ -156,33 +161,43 @@ def update(self, size, csize, unique):
         if unique:
             self.usize += csize
 
-    def print_(self, label, cache):
-        buf = str(self) % label
-        buf += "\n"
-        buf += str(cache)
-        return buf
-
-    def __str__(self):
-        return format(self, """\
+    summary = """\
                        Original size      Compressed size    Deduplicated size
-%-15s {0.osize:>20s} {0.csize:>20s} {0.usize:>20s}""")
+{label:15} {stats.osize_fmt:>20s} {stats.csize_fmt:>20s} {stats.usize_fmt:>20s}"""
+    def __str__(self):
+        return self.summary.format(stats=self, label='This archive:')
+
+    def __repr__(self):
+        fmt = "<{cls} object at {hash:#x} ({self.osize}, {self.csize}, {self.usize})>"
+        return fmt.format(cls=type(self).__name__,
```

Comment:
sorry, which wrap?

if you mean the fmt bit, it was just to clarify the code..